src/main/java/com/uid2/shared/store/reader/RotatingS3KeyProvider.java

```diff
@@ -23,7 +28,7 @@ public class RotatingS3KeyProvider implements StoreReader<Map<Integer, S3Key>> {
     ScopedStoreReader<Map<Integer, S3Key>> reader;
 
     private static final Logger LOGGER = LoggerFactory.getLogger(RotatingS3KeyProvider.class);
-    public final Map<Integer, List<S3Key>> siteToKeysMap = new HashMap<>();
+    Map<Integer, List<S3Key>> siteToKeysMap = new ConcurrentHashMap<>();
```

Comment:
Why is this changed to a concurrentHashMap?

Reply:
I am a little bit worried that there would be a concurrency issue,  such as simultaneous updates to the siteId map keys and reading keys occurring at the same time. Can revert this back to hashMap if thats not a problem